### ISSUE TITLE
Allow to skip RTD builds on CI

### ISSUE THREAD

**Issue by hamogu** [astropy/astropy#16176]
### What is the problem this feature will solve?

RTD builds takes a long time, so we don't want to hammer RTD to build when it's not needed. On GH Actions we use `[skip ci]` for this purpose, but we have no equivalent mechanism for RTD yet.

### Additional context

 https://github.com/astropy/astropy/issues/16161#issuecomment-1984178973 provides a way to implement the same feature for RTD.

**Comment by pllim** [comment]
I replied in https://github.com/astropy/astropy/issues/16161#issuecomment-1984222785 but cross-posting here:

I actually implemented that for another package here:

https://github.com/spacetelescope/jdaviz/blob/c731eef855a0a65a6172fe53fca499ff6a98b3d9/.readthedocs.yaml#L13

However, implementing it for `astropy` needs more thoughts:

* We used to have a `[docs only]` directive that is basically what `[skip ci]` does now (skips the Actions but not RTD). I (ab)use this feature sometimes.
* To keep the above, we can introduce `[skip rtd]` like I did with the other package. Problem is because it is a completely new directive, most of the time I forgot to use it or too lazy to type more stuff into the commit. So it is not used a lot in the end.

**Comment by mhvk** [comment]
In principle, could move the RTD run to weekly cron, perhaps unless actual .rst files are touched (with perhaps an auto-checker that no changelog fragment ever contains a reference). Downside is that unless we take some special measures, it basically moves work from PR submitters/reviewers to, effectively, @pllim. So, we'd need some special measures! Someone who gets notified if RTD breaks and fixes it.

**Comment by pllim** [comment]
> move the RTD run to weekly cron

I don't think that is a feature at all. There is no cron option over at RTD. Also, it would be a shame to not take advantage of their nice PR build with HTML preview.

**Comment by mhvk** [comment]
> Also, it would be a shame to not take advantage of their nice PR build with HTML preview.

Ah, yes, see I always forget something - the HTML preview is indeed fantastic! 

So, that leaves an option to disable the docs, which, as you say, most people will forget, or some form of (semi-)automatic inclusion of the docs build as needed. E.g., we could perhaps trigger on the `Docs` label?

**Comment by pllim** [comment]
Re: trigger on label -- Would have to ask RTD if they grab PR labels. If so, maybe possible. However, even then, we rely on maintainers to comb through the diff carefully and able to determine whether any arbitrary diff would affect the doc build or not. Is the risk too high? 

p.s. Is the current problem that bad? When I wait for RTD to build for my PR, I have a million other things I can work on. Does anyone just sit in front of the screen staring at the RTD build status?

**Comment by pllim** [comment]
I can definitely add the `[rtd skip]` and `[skip rtd]` if you find it helpful. I didn't but does not mean others won't.

**Comment by pllim** [comment]
Or if you want `[ci skip]` to skip everything including RTD, I can also do that. But like I said, we would lose the `[docs only]` equivalent functionality.

**Comment by pllim** [comment]
BTW I found an old issue that I opened but forgotten... 😆 

* https://github.com/astropy/astropy/issues/14152

**Comment by pllim** [comment]
I forgotten we left this issue linger. After 2 years, sometimes I actually find `[ci skip]` not skipping RTD useful, like testing doc only PRs. So I will close this as won't fix for now...

**Comment by humitos** [comment]
@pllim Hi 👋🏼 -- We improved Read the Docs' automation rules and now they support commit filters to decide whether or not to trigger the build. The feature is already deployed, but the documentation is not (you can find it in the PR at https://github.com/readthedocs/readthedocs.org/pull/13021 and the blog post at https://github.com/readthedocs/website/pull/405

We will be publishing them today.

Let me know if that covers your use case 👍🏼 . We appreciate any feedback you may have.
